test/TwigExtensionFactoryTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-twigrenderer for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-twigrenderer/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace ZendTest\Expressive\Twig;
+
+use PHPUnit\Framework\TestCase;
+use Prophecy\Prophecy\ProphecyInterface;
+use Psr\Container\ContainerInterface;
+use Zend\Expressive\Helper\ServerUrlHelper;
+use Zend\Expressive\Helper\UrlHelper;
+use Zend\Expressive\Twig\Exception\InvalidConfigException;
+use Zend\Expressive\Twig\TwigExtension;
+use Zend\Expressive\Twig\TwigExtensionFactory;
+
+class TwigExtensionFactoryTest extends TestCase
+{
+    /**
+     * @var ContainerInterface|ProphecyInterface
+     */
+    private $container;
+
+    /**
+     * @var ServerUrlHelper|ProphecyInterface
+     */
+    private $serverUrlHelper;
+
+    /**
+     * @var UrlHelper|ProphecyInterface
+     */
+    private $urlHelper;
+
+    public function setUp()
+    {
+        $this->container       = $this->prophesize(ContainerInterface::class);
+        $this->serverUrlHelper = $this->prophesize(ServerUrlHelper::class);
+        $this->urlHelper       = $this->prophesize(UrlHelper::class);
+    }
+
+    public function testRaisesExceptionForMissingServerUrlHelper()
+    {
+        $this->container->has(ServerUrlHelper::class)->willReturn(false);
+
+        $this->expectException(InvalidConfigException::class);
+        $this->expectExceptionMessage(sprintf(
+            'Missing required `%s` dependency.',
+            ServerUrlHelper::class
+        ));
+
+        (new TwigExtensionFactory())($this->container->reveal());
```

Comment:
Usually we create property `$this->factory = new TwigExtensionFactory()` as this is basically used in every test, but it's fine as it is.